client/src/main/java/com/microsoft/durabletask/TaskOrchestrationExecutor.java

```diff
@@ -1271,6 +1271,24 @@ public V await() {
                         "The orchestrator is blocked and waiting for new inputs. This Throwable should never be caught by user code.");
             }
 
+            private boolean processNextEvent() {
+                try {
+                    return ContextImplTask.this.processNextEvent();
+                } catch (OrchestratorBlockedException | ContinueAsNewInterruption exception) {
+                    throw exception;
+                } catch (Exception e) {
+                    // ignore
+                    /**
+                     * the idea is that for any exception thrown here it should be handled
+                     * in the do-while loop, the same exception should also be obtained when calling
+                     * {code#future.get()} in the do-while loop, and it will be handled correctly
+                     * in the catch block in the do-while loop.
+                     */
+                }
+                // Any exception happen we return true so that we will enter to the do-while block for the last time.
```

Comment:
some nits on phrasing:

```suggestion
                    /**
                     * We ignore the exception. Any Durable Task exceptions thrown here can be obtained when calling
                     * {code#future.get()} in the implementation of 'await'. We defer to that loop to handle the exception.
                     */
                }
                // On an exception, we return true to the do-while block one last time.
```